src/PostgREST/Request/ApiRequest.hs

```diff
@@ -180,12 +168,18 @@ data ApiRequest = ApiRequest {
 
 -- | Examines HTTP request and translates it into user intent.
 userApiRequest :: AppConfig -> DbStructure -> Request -> RequestBody -> Either ApiRequestError ApiRequest
-userApiRequest conf@AppConfig{..} dbStructure req reqBody
+userApiRequest conf dbStructure req reqBody =
+  userApiRequest' conf dbStructure req reqBody =<< first QPError (QueryParams.parse (rawQueryString req))
+
+userApiRequest' :: AppConfig -> DbStructure -> Request -> RequestBody -> QueryParams.QueryParams -> Either ApiRequestError ApiRequest
+userApiRequest' conf@AppConfig{..} dbStructure req reqBody querystring
+  | isTargetingProc && method == "POST" && not (L.null qsParams) = Left $ ParseRequestError "" ""
+  | not isTargetingProc && not (L.null qsParams) =
+      Left $ ParseRequestError "Unexpected param or filter missing operator" ("Failed to parse " <> show qsParams)
```

Comment:
This actually looks like a feature "Improved error messages" for those cases. Although I don't understand why we have `ParseRequestError "" ""` in there - is this error actually reported somewhere?

Not sure whether it would be possible to split those changes into a separate commit, together with the changed test output, because this is not a refactor anymore.

Reply:
This is difficult to split from the rest of the refactor, as catching this error case here (params provided although no params are expected as no proc is being called) significantly simplifies the remaining logic of parsing the query string. Providing a better error message is a side effect that is hard to avoid in this case.